src/Symfony/DependencyInjection/AsyncAwsExtension.php

```diff
@@ -0,0 +1,113 @@
+            }
+
+            $config = array_merge($defaultConfig, $data);
+            if ($config['register_service']) {
+                $usedServices[$name] = $client;
+                $this->addServiceDefinition($container, $name, $config, $client);
+            } else {
+                $usedServices[$name] = null;
+            }
+        }
+
+        return $usedServices;
+    }
+
+    private function registerInstalledServices(ContainerBuilder $container, array $config, array $usedServices): array
+    {
+        if (!$config['register_service']) {
+            return $usedServices;
+        }
+
+        unset($config['services']);
+        $availableServices = AwsPackagesProvider::getAllServices();
+        foreach ($availableServices as $name => $data) {
+            if (\array_key_exists($name, $usedServices)) {
+                continue;
+            }
+
+            $client = $data['client'];
+            if (!class_exists($client)) {
+                continue;
+            }
+
+            $usedServices[$name] = $client;
+            $this->addServiceDefinition($container, $name, $config, $client);
+        }
+
+        return $usedServices;
+    }
+
+    private function addServiceDefinition(ContainerBuilder $container, string $name, array $config, string $clientClass): void
+    {
+        if (\array_key_exists('logger', $config)) {
+            $logger = $config['logger'] ? new Reference($config['logger']) : null;
+        } else {
+            // Use default Symfony logger unless explicitly set to null.
+            $logger = new Reference('logger', ContainerInterface::NULL_ON_INVALID_REFERENCE);
+        }
+
+        $definition = new Definition($clientClass);
+        $definition->addArgument($config['config']);
+        $definition->addArgument(isset($config['credential_provider']) ? new Reference($config['credential_provider']) : null);
+        $definition->addArgument(isset($config['http_client']) ? new Reference($config['http_client']) : null);
+        $definition->addArgument($logger);
+        $container->setDefinition(sprintf('async_aws.service.%s', $name), $definition);
+    }
+
+    private function autowireServices(ContainerBuilder $container, array $usedServices): void
+    {
+        $awsServices = AwsPackagesProvider::getAllServices();
+        foreach ($usedServices as $name => $client) {
+            if (null === $client) {
+                // This client is disabled.
+                continue;
+            }
+
+            $serviceId = sprintf('async_aws.service.%s', $name);
+            if (isset($awsServices[$name])) {
+                $container->setAlias($client, $serviceId);
+
+                continue;
+            }
+
+            // Assert: Custom name
+            $container->registerAliasForArgument($serviceId, $client, $name);
```

Comment:
What's about also adding `$container->registerAliasForArgument($serviceId, $client)` for the default service (`sqs`, `s3`, ..).

It would allow out of the box
```shell
$ composer req async-aws/sqs
```
```yaml
async_aws:
    config:
        region: eu-central-1
```

```php
public function __construct(SqsClient $myLittlerClient) {
}
```

Reply:
Isn’t that already the case? I’m sure it is. That is at least the intention.

Reply:
Oh nice !!.. I thougth that the last argument of `registerAliasForArgument` was the name of the property to bind..